docs/writing-rules/pattern-syntax.md

```diff
@@ -421,6 +424,20 @@ It will not match the following:
 import foo
 ```
 
+### Statements as expressions
+
```

Comment:
I would put the little paragraph on expr vs statements here.
Then you could add:
"Note that in some programming languages such as Python, which does not use semicolons as a separator or terminator, things are even more confusing, but indentations in Python matters, and a newline after foo() is really the same than foo(); in other programming languages such as C."